packages/components/pie-modal/src/index.ts

```diff
@@ -359,85 +356,93 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
     }
 
     /**
-     * Renders the modal inner content and footer of the modal.
+     * Renders the modal footer if a leading action is provided.
+     * Additionally renders the supporting action if it is provided.
+     *
      * @private
      */
-    private renderModalContentAndFooter (): TemplateResult {
-        const hasFooterLeadingAction = Boolean(this.leadingAction?.text);
-
-        const scrollContainerClasses = {
-            'c-modal-scrollContainer': true,
-            'c-modal-content': true,
-            'c-modal-content--scrollable': true,
-            'c-modal-hasFooterActions': hasFooterLeadingAction,
-        };
+    private renderModalFooter () : TemplateResult | typeof nothing {
+        if (!this.leadingAction?.text) {
+            if (this.supportingAction?.text) {
+                console.warn('You cannot have a supporting action without a leading action. If you only need one button then use a leading action instead.');
+            }
+            return nothing;
+        }
 
         return html`
-            <article class=${classMap(scrollContainerClasses)}>
+            <footer class="c-modal-footer" data-test-id="pie-modal-footer">
+                ${this.renderLeadingAction()}
+                ${this.renderSupportingAction()}
+            </footer>`;
+    }
+
+    /**
+     * Renders the loading spinner if `isLoading` is true.
+     * @private
+     */
+    private renderLoadingSpinner () : TemplateResult | typeof nothing {
+        if (!this.isLoading) {
+            return nothing;
+        }
+
+        return html`<pie-spinner size="xlarge" variant="secondary"></pie-spinner>`;
+    }
+
+    /**
+     * Renders the modal inner content and footer of the modal.
+     * @private
+     */
+    private renderModalContentAndFooter () : TemplateResult {
+        return html`
+            <article class="c-modal-scrollContainer c-modal-content c-modal-content--scrollable">
                 <div class="c-modal-contentInner" data-test-id="modal-content-inner">
                     <slot></slot>
                 </div>
-                ${this.isLoading ? html`<pie-spinner size="xlarge" variant="secondary"></pie-spinner>` : nothing}
+                ${this.renderLoadingSpinner()}
             </article>
-            ${hasFooterLeadingAction ? html`
-                <footer class="c-modal-footer" data-test-id="pie-modal-footer">
-                    ${this.renderLeadingAction()}
-                    ${this.renderSupportingAction()}
-                </footer>` : nothing}`;
+            ${this.renderModalFooter()}`;
     }
 
     public render () {
-        const {
-            aria,
-            hasBackButton,
-            hasStackedActions,
-            heading,
-            headingLevel,
-            isDismissible,
-            isFooterPinned,
-            isFullWidthBelowMid,
-            isLoading,
-            leadingAction,
-            position,
```

Comment:
As discussed offline, I like this change however it does deviate from the way we reference properties in other components.

I think this could be a nice change but I would prefer that we stay consistent. So I would suggest either chat with the team to get some agreement on which pattern to go with, and if that requires changes lets create a ticket and tackle them all. 

Perhaps there is some kind of linting we can add in the future around this to enforce whichever pattern we choose.